SU-HSDO/suhumsci
# STN-777: Color band sidebar refinement

# READY FOR REVIEW

## Summary
We want the Color Band to be visible full width in the sidebar left/right sidebar regions unless a block extends over the element.

This work also adds a utility class called `hb-sidebar-block` which applies the existing styles that sidebar group blocks already receive.  This allows the same styles to be added onto blocks within the layout builder for the sidebar for things such as custom blocks, which are edge cases.

JIRA Card: https://sparkbox.atlassian.net/browse/STN-777

## Need Review By (Date)
6/4

## Urgency
medium

## Steps to Test
1. npm run test
2. To test this item, you'll need to add a Color Band on a page that has menu sidebars or you'll want to add sidebar elements to a page with color bands you are testing manually. A page I used for testing was http://sparkbox-sandbox.suhumsci.loc/qa/qa-embeds.
3. You'll want to test against menu blocks on each sidebar, as well as adding other block types using examples within the JIRA Card such as "Group Blocks: Content", then adding custom blocks within it that are listed such as "about Stanford", etc.
4. We've also added a utility class in this work which gives the same padding and the border that is around the group blocks. This can be tested by adding a custom-created block to the sidebar, use the "+ Create Custom Block" button that is featured at the top of the Layout Builder block sidebar menu. Then using "manage attributes" on the block add the `hb-sidebar-block` class on that block. This should resemble the same look as any of the group blocks that are added by themselves.
![image](https://user-images.githubusercontent.com/9355698/120701262-a18a8580-c480-11eb-8939-4f38a0c5eb9a.png)

- [x] Verify `hb-well` works with color band when applied to a sidebar group block via Layout Builder.
- [x] Verify these tests work with each theme and color pairings.
- [x] Verify the utility class has also been added to the Group Block Utility Class section of our documentation.


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
